app/main.py

```diff
@@ -1,3 +1,15 @@
 def create_dict(*args):
-    # write your code here
-    pass
+    out = {}
+    for index, key in enumerate(args):
+        if isinstance(key, (int, float, str, bool)) or callable(key) or None:
+            out[key] = index
+            continue
+        if isinstance(key, tuple):
+            if not any([isinstance(key_key, (set, list)) for key_key in key]):
+                out[key] = index
+                continue
+        print(f"Cannot add {key} to the dict!")
+    return out
+
+    # Is it possible to leave such a construction, without (if elif else)?
+    # it seems to me that in this case it increases the size of the code
```

Comment:
It adds only 1 row of code, but it becomes more readable and code reviewer doesn't waste time to understand all this if+continue